### ISSUE TITLE
Look into continuous integration and unit testing

### ISSUE THREAD

**Issue by gbear605** [Nashoba-Robotics/Nashoba-Robotics2017#14]
In 2016 we made [minimal use of continuous integration](https://travis-ci.org/Nashoba-Robotics/Nashoba-Robotics2016), but it didn't really help. To be more useful in the future, it would need to have more unit tests than the basic ones that I implemented. Those unit tests would require a version of WPILib that doesn't require the presence of the robot. I know that @Team254 has done this in the past, so it might be possible to use their version, or another team's version. If not, we will need to create a version ourselves. I suspect that that difficulty combined with the difficulty of writing unit tests for an environment like FRC will outweigh the benefit of unit tests.

**Comment by Ipsum** [comment]
I am 100% for testing always but without a decent test harness I would have to vote to make this lower priority. We do need to figure out some way of controlling the quality of code even under pressure though.

**Comment by arlevin** [comment]
@Ipsum Agreed.  It's the 2nd lowest item.  @gbear605 please let us know if you disagree

**Comment by gbear605** [comment]
I agree with it being very low priority.
